Common/Python/DataConsolidatorPythonWrapper.cs

```diff
@@ -0,0 +1,107 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using Python.Runtime;
+using QuantConnect.Data;
+using QuantConnect.Data.Consolidators;
+
+namespace QuantConnect.Python
+{
+    /// <summary>
+    /// Provides an Data Consolidator that wraps a <see cref="PyObject"/> object that represents a custom Python consolidator
+    /// </summary>
+    public class DataConsolidatorPythonWrapper : IDataConsolidator
+    {
+        private readonly dynamic _consolidator;
+
+        /// <summary>
+        /// Gets the most recently consolidated piece of data. This will be null if this consolidator
+        /// has not produced any data yet.
+        /// </summary>
+        public IBaseData Consolidated 
+        { 
+            get { using (Py.GIL()) {return _consolidator.Consolidated;} }
+        }
+
+        /// <summary>
+        /// Gets a clone of the data being currently consolidated
+        /// </summary>
+        public IBaseData WorkingData 
+        { 
+            get { using (Py.GIL()) {return _consolidator.WorkingData;} }
+        }
+
+        /// <summary>
+        /// Gets the type consumed by this consolidator
+        /// </summary>
+        public Type InputType
+        {
+            get { using (Py.GIL()) {return _consolidator.InputType;} }
+        }
+
+        /// <summary>
+        /// Gets the type produced by this consolidator
+        /// </summary>
+        public Type OutputType 
+        { 
+            get { using (Py.GIL()) {return _consolidator.OutputType;} } 
+        }
+
+        /// <summary>
+        /// Event handler that fires when a new piece of data is produced
+        /// </summary>
+        public event DataConsolidatedHandler DataConsolidated;
```

Comment:
Think we are missing to hock up this event handler with the underlying python event triggering and adding a new interface for `RegisterIndicator` which supports using a python based consolidator, basically wrapping it with this class

Reply:
- Unit test added for event triggering
- Register indicator implemented and tested
- Regression algo now uses register indicator